Silicon/NXP/iMX6Pkg/Drivers/PlatformSmbiosDxe/PlatformSmbiosDxe.c

```diff
@@ -497,6 +498,98 @@ GetImx6SerialNumber (
   return ProcessorSerialNumber;
 }
 
+// {72096f5b-2ac7-4e6d-a7bb-bf947d673415}
+EFI_GUID ProvisioningGuid =
+{ 0x72096f5b, 0x2ac7, 0x4e6d, { 0xa7, 0xbb, 0xbf, 0x94, 0x7d, 0x67, 0x34, 0x15 } };
+
+STATIC CONST CHAR16  mSystemSerialNumber[] = L"SmbiosSystemSerialNumber";
+STATIC CONST CHAR8   SystemSerialNumber[] = "SerialNumberFromUEFIVars\0";
+
+STATIC CONST CHAR16  mSystemSkuNumber[] = L"SmbiosSystemSkuNumber";
+STATIC CONST CHAR8   SystemSkuNumber[] = "ActualSkuFromUEFIVars\0";
+
+EFI_STATUS
+RetrieveSmbiosVariable (
+  CONST CHAR16 *VariableName,
+  CHAR16 *SmbiosEntry,
+  UINTN *SmbiosEntryLen,
+  UINT32 *SmbiosRecordLen
+  )
+{
+  CHAR8                 *Data;
+  UINTN                 DataSize;
+  EFI_STATUS            Status;
+
+  DataSize = 0;
+  Status = gRT->GetVariable (
+                  (CHAR16 *) VariableName,
+                  &ProvisioningGuid,
+                  NULL,
+                  &DataSize,
+                  NULL
+                  );
+
+  if (Status == EFI_BUFFER_TOO_SMALL) {
+    Data = AllocatePool (DataSize);
+    if (Data == NULL) {
+      return EFI_OUT_OF_RESOURCES;
+    }
+
+    Status = gRT->GetVariable (
+                    (CHAR16 *) VariableName,
+                    &ProvisioningGuid,
+                    NULL,
+                    &DataSize,
+                    Data
+                    );
+
+      (VOID)UnicodeSPrintAsciiFormat (SmbiosEntry,
+                                  sizeof (CHAR16) * SMBIOS_STRING_MAX_LENGTH,
+                                  Data
+                                  );
+
+      *SmbiosEntryLen = StrLen (SmbiosEntry);
+      *SmbiosRecordLen += *SmbiosEntryLen + 1;
+
+      FreePool (Data);
+      Status = EFI_SUCCESS;
+    }
+  return Status;
+}
+
+EFI_STATUS
+StoreSmbiosVariable (
+  CONST CHAR16 *VariableName,
+  CONST CHAR8 *VariableString
+  )
+{
+  EFI_STATUS            Status;
+  DEBUG ((DEBUG_ERROR, "%a: Setting SmbiosVariable\n", __FUNCTION__));
+
+  Status = gRT->SetVariable (
+                  (CHAR16 *)VariableName,
+                  &ProvisioningGuid,
+                  EFI_VARIABLE_NON_VOLATILE | EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS,
+                  AsciiStrLen(VariableString)+1,
+                  (VOID *)VariableString
+                  );
+
+  return Status;
+}
+
+EFI_STATUS

```

Comment:
This is test code that's required until we switch to full persistent AuthVars. Otherwise we dont have any values in UEFI vars to pull out and plumb into the SMBIOS tables after every reset.

Reply:
Right, but this code should definitely not go into the imx branch later. I'd prefer it to not even go into this provisioning branch but it sounds like we need it to demo some flow.

Add a comment to ensure we remove this before merging into the main branch.

Reply:
Added an all caps TODO: REMOVE THIS FUNCTION in the function description comment.